resources/scss/_common.scss

```diff
@@ -0,0 +1,321 @@
+  justify-content: center;
+}
+
+*,
+*::after,
+*::before {
+  margin: 0;
+  padding: 0;
+  box-sizing: inherit;
+  color: $text-color-primary;
+}
+
+a {
+  color: $text-color-primary;
+  text-decoration: none;
+  user-select: none;
+}
+
+html,
+body {
+  margin: 0 auto;
+  font-family: 'Dosis';
+  font-weight: $app-font-weight;
+  box-sizing: border-box;
+  font-size: 0.75rem;
+  @include break-point(tablet) {
+    font-size: 1rem;
+  }
+}
+
+.btn {
+  background: $background-color-secondary;
+  border: 0;
+  padding: $app-btn-padding;
+  color: $text-color-secondary;
+  font-family: 'Dosis';
+  text-align: center;
+  font-size: 1rem;
+}
+
+.nav {
+  display: flex;
+  height: $header-height;
+  box-shadow: 0 8px 6px -3px #eeeeee;
+  @include app-header-padding-changes;
+
+  &__logo {
+    margin-left: 1vw;
+    width: 29vw;
+    padding: 9.5px 0;
+    @include break-point(tablet) {
+      margin-left: 0;
+      width: 16vw;
+    }
+  }
+  &__page {
+    @extend .flex-row;
+    &-link {
+      padding: 9.5px 0;
+      margin-left: 9vw;
+      display: none;
+      color: #736a6d;
+      align-self: flex-end;
+      @include break-point(tablet) {
+        display: flex;
+      }
+    }
+  }
+
+  &__end {
+    @extend .flex-column;
+    flex: 0 1 12vw;
+    @include break-point(tablet) {
+      flex-basis: 16vw;
+    }
+    @include break-point(desktop) {
+      flex-basis: 12vw;
+    }
+
+    &-register {
+      @extend .flex-row;
+      justify-content: flex-end;
+      height: 34%;
+      display: none;
+      @include break-point(tablet) {
+        display: flex;
+      }
+
+      &-link {
+        font-size: 0.7rem;
+        margin-left: 1.5vw;
+      }
+    }
+    &-cart {
+      @extend .flex-row;
+      @extend .flex-center;
+      background-color: #eeeeee;
+
+      &-icon {
+        position: relative;
+      }
+      &-logo {
+        height: 0.36 * $header-height;
+        fill: $background-color-secondary;
+      }
+      &-count {
+        display: none;
+        @include break-point(tablet) {
+          display: block;
+        }
+        &-xs {
+          position: absolute;
+          top: -15%;
+          left: 50%;
+          transform: translate(-50%, 0);
+          background: #50dae3;
+          color: white;
+          height: 1.5rem;
+          width: 1.5rem;
+          text-align: center;
+          border-radius: 1rem;
+          @extend .flex-row;
+          @extend .flex-center;
+          @include break-point(tablet) {
+            display: none;
+          }
+        }
+      }
+    }
+  }
+}
+
+// toggle menu for mobile
+.menu-toggle {
+  display: block;
+  position: relative;
+  z-index: 1;
+  -webkit-user-select: none;
+  user-select: none;
+  align-self: center;
+  @include break-point(tablet) {
+    display: none;
+  }
+
+  input {
+    display: block;
+    width: 40px;
+    height: 32px;
+    position: absolute;
+    top: -7px;
+    left: -5px;
+    cursor: pointer;
+    opacity: 0;
+    z-index: 2;
```

Comment:
Keep fixed number of z-indexes values. And store them per functional requirement in variables. E.g. 
0 -> normal
1 -> layovers
2 -> modals